src/test/java/org/cactoos/map/MapEnvelopeTest.java

```diff
@@ -203,4 +207,103 @@ public void mapContainsValueFalse() {
         );
     }
 
+    @Test
+    public void mapEqualsToItself() {
+        final MapOf<String, String> map =
+            new MapOf<String, String>(new MapEntry<>("key", "value"));
+        MatcherAssert.assertThat(
+            "Map doesn't equal to itself",
+            map,
+            new IsEqual<>(map)
+        );
+    }
+
+    @Test
+    public void mapNotEqualsToAnotherClass() {
+        final MapOf<String, String> map =
+            new MapOf<String, String>(new MapEntry<>("key1", "value1"));
+        MatcherAssert.assertThat(
+            "Map equals to an instance of another type",
+            map,
+            new IsNot<>(
+                new IsEqual<>("Totally different type")
+            )
+        );
+    }
+
+    @Test
+    public void mapEqualsToMapWithSameEntries() {
+        final MapEntry<String, String> entry =
+            new MapEntry<>("key2", "value2");
+        MatcherAssert.assertThat(
```

Comment:
@proshin-roman To enforce value comparison I'd create a new instance of `MapEntry` with the same key / value pair `("key2", "value2")` and use it to check if two maps with the same key / value are equals. What do you think?

Reply:
@paulodamaso Well, ok, it makes sense, I will fix it.